apollo-router/src/uplink/mod.rs

```diff
@@ -264,68 +300,100 @@ where
     Response: Send + Debug + 'static,
+    SomethingElse: Send + Debug + 'static,
 {
     let query = query_name::<Query>();
     for url in urls {
         let now = Instant::now();
         match http_request::<Query>(client, url.as_str(), request_body).await {
-            Ok(response) => {
-                let response = response.data.map(Into::into);
-                match &response {
-                    None => {
-                        tracing::info!(
-                            histogram.apollo_router_uplink_fetch_duration_seconds =
-                                now.elapsed().as_secs_f64(),
-                            query,
-                            url = url.to_string(),
-                            "kind" = "uplink_error",
-                            error = "empty response from uplink",
-                        );
-                    }
-                    Some(UplinkResponse::New { .. }) => {
-                        tracing::info!(
-                            histogram.apollo_router_uplink_fetch_duration_seconds =
-                                now.elapsed().as_secs_f64(),
-                            query,
-                            url = url.to_string(),
-                            "kind" = "new"
-                        );
-                        return Ok(response.expect("we are in the some branch, qed"));
-                    }
-                    Some(UplinkResponse::Unchanged { .. }) => {
-                        tracing::info!(
-                            histogram.apollo_router_uplink_fetch_duration_seconds =
-                                now.elapsed().as_secs_f64(),
-                            query,
-                            url = url.to_string(),
-                            "kind" = "unchanged"
-                        );
-                        return Ok(response.expect("we are in the some branch, qed"));
-                    }
-                    Some(UplinkResponse::Error { message, code, .. }) => {
-                        tracing::info!(
-                            histogram.apollo_router_uplink_fetch_duration_seconds =
-                                now.elapsed().as_secs_f64(),
-                            query,
-                            url = url.to_string(),
-                            "kind" = "uplink_error",
-                            error = message,
-                            code
-                        );
-                        return Ok(response.expect("we are in the some branch, qed"));
+            Ok(response) => match response.data.map(Into::into) {
+                None => {
+                    tracing::info!(
+                        histogram.apollo_router_uplink_fetch_duration_seconds =
+                            now.elapsed().as_secs_f64(),
+                        query,
+                        url = url.to_string(),
+                        "kind" = "uplink_error",
+                        error = "empty response from uplink",
```

Comment:
Could you check that `"uplink_error"` in metrics is not in a debug version ? 

I'm used to write 
`"kind" = %"uplink_error"` to force the display formatting of that attribute. But I'm not sure it's useful or not

Reply:
If it was already there, then let's keep it. FYI the `%` or `?` is documented in the [tracing macro documentation](https://docs.rs/tracing/latest/tracing/index.html#recording-fields) it's to use the `Debug` or `Display` implementation of the attribute :)

Reply:
Ah gotcha. But I think this is just literally the string `"uplink_error"`?